src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/SearchObject.cs

```diff
@@ -233,6 +235,51 @@ private void CacheBackgroundData(IPackageSearchMetadata packageSearchMetadata, b
             {
                 memoryCacheItem.UpdateSearchMetadata(packageSearchMetadata);
             }
+
+            Assumes.NotNull(packageSearchMetadata);
+            NuGetRemoteFileService.AddIconToCache(packageSearchMetadata.Identity, packageSearchMetadata.IconUrl);
+
+            string? packagePath = (packageSearchMetadata as LocalPackageSearchMetadata)?.PackagePath ??
+                    (packageSearchMetadata as ClonedPackageSearchMetadata)?.PackagePath;
+
+            if (packagePath != null)
+            {
+                LicenseMetadata? licenseMetadata = (packageSearchMetadata as LocalPackageSearchMetadata)?.LicenseMetadata ??
+                    (packageSearchMetadata as ClonedPackageSearchMetadata)?.LicenseMetadata;
+                if (licenseMetadata != null)
+                {
+                    NuGetRemoteFileService.AddLicenseToCache(
+                        packageSearchMetadata.Identity,
+                        CreateEmbeddedLicenseUri(packagePath, licenseMetadata));
+                }
+            }
```

Comment:
refactoring suggestion..
```suggestion
             IPackageSearchMetadata localOrClonedPackageSearchMetadata = (packageSearchMetadata as LocalPackageSearchMetadata) ??
                    (packageSearchMetadata as ClonedPackageSearchMetadata);

            if (localOrClonedPackageSearchMetadata?.PackagePath != null)
            {
                    NuGetRemoteFileService.AddLicenseToCache(
                        packageSearchMetadata.Identity,
                        CreateEmbeddedLicenseUri(localOrClonedPackageSearchMetadata.PackagePath, licenseMetadata));
                }
            }
```

Reply:
EDITED suggestion: Sorry, with fresh eyes I see I had missed the License metadata altogether 😄

Reply:
I had added PackagePath to IPackageSearchMetadata initially, but based on @zivkan's feedback, i moved the new API to only the subclasses that it applied to.
Given your first line of the refactor depends on undoing that, i'm resolving. Feel free to open additional feedback given that.